### PR TITLE
[Feature][Transform] Sql transform support array and split function

### PR DESCRIPTION
**Pull request by CosmosNi**
selectItem support array and split func

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Review by Hisoka-X — APPROVED** [review]
LGTM if ci passes. Thanks @CosmosNi !

**Review by hailin0 — APPROVED** [review]

File: seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/functions/SystemFunction.java [hunk 1]
```diff
@@ -70,11 +68,13 @@ public static Object nullif(List<Object> args) {
         return v1;
     }
 
-    public static Object array(List<Object> args) {
+    public static String[] array(List<Object> args) {
```
Comment (hailin0) [inline seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/functions/SystemFunction.java:71]: Why does it return a fixed string[]?

Example:
Array(1, 2, 3) returns Integer[]
Array(1L, 2, 3) returns Long[]
Array(1L, 1.1F, 2.1D) returns Double[]
Array(true, false, true) returns Boolean[]
Array(true, "aaa", 1) returns String[]
Array(1, true, 3L) returns String[]

Comment (hailin0) [reply seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/functions/SystemFunction.java:71]: Of course, you can merge this pr and submit a new PR to optimize this function

Comment (CosmosNi) [reply seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/functions/SystemFunction.java:71]: case ZetaSQLFunction.ARRAY:
       return ArrayType.STRING_ARRAY_TYPE;

Then we need to dynamically generate the returned ArrayType.

Comment (Hisoka-X) [reply seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/functions/SystemFunction.java:71]: > Then we need to dynamically generate the returned ArrayType.

+1, we can reuse https://github.com/apache/seatunnel/blob/d2576ff13112b0f845930374632a959cf29e7eb9/seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/TypeUtil.java#L23 to check and convert type.

**Pull request merged**